tests/integration/standard/test_query.py

```diff
@@ -469,6 +466,38 @@ def test_prepare_on_all_hosts(self):
         finally:
             clus.shutdown()
 
+    def test_prepare_batch_statement(self):
+        """
+        Test to validate a prepared statement used inside a batch statement is correctly handled
+        by the driver
+
+        @since 3.10
+        @jira_ticket PYTHON-706
+        @expected_result queries will have to re-prepared on hosts that aren't the control connection
+        and the batch statement will be sent.
+        """
+        white_list = ForcedHostSwitchPolicy()
+        clus = Cluster(
+            load_balancing_policy=white_list,
+            protocol_version=PROTOCOL_VERSION, prepare_on_all_hosts=False,
+            reprepare_on_up=False)
+        try:
+            session = clus.connect(wait_for_all_pools=True)
+
+            insert_statement = session.prepare("INSERT INTO test3rf.test (k, v) VALUES  (?, ?)")
+
+            # This is going to query a host where the query
+            # is not prepared
+            batch_statement = BatchStatement(consistency_level=ConsistencyLevel.ONE)
+            batch_statement.add(insert_statement, (1, 2))
+            session.execute(batch_statement)
+            select_results = session.execute("SELECT * FROM test3rf.test WHERE k = 1")
+            first_row = select_results[0][:2]
+            self.assertEqual((1, 2), first_row)
+
+        finally:
+            clus.shutdown()
```

Comment:
You can avoid doing this in a `try/finally` by running `self.addCleanup(clus.shutdown)` after constructing `clus`.